lib/ansible/modules/crypto/entrust/ecs_certificate.py

```diff
@@ -0,0 +1,890 @@
+        dropdown3=dict(type='str'),
+        dropdown4=dict(type='str'),
+        dropdown5=dict(type='str'),
+    )
+
+
+def ecs_certificate_argument_spec():
+    return dict(
+        backup=dict(type='bool', default=False),
+        force=dict(type='bool', default=False),
+        path=dict(type='path', required=True),
+        chain_path=dict(type='path'),
+        tracking_id=dict(type='int'),
+        remaining_days=dict(type='int', default=30),
+        request_type=dict(type='str', default='NEW', choices=['NEW', 'RENEW', 'REISSUE']),
+        cert_type=dict(type='str', choices=['STANDARD_SSL',
+                                            'ADVANTAGE_SSL',
+                                            'UC_SSL',
+                                            'EV_SSL',
+                                            'WILDCARD_SSL',
+                                            'PRIVATE_SSL',
+                                            'PD_SSL',
+                                            'CODE_SIGNING',
+                                            'EV_CODE_SIGNING',
+                                            'CDS_INDIVIDUAL',
+                                            'CDS_GROUP',
+                                            'CDS_ENT_LITE',
+                                            'CDS_ENT_PRO',
+                                            'SMIME_ENT',
+                                            ]),
+        csr=dict(type='str'),
+        subject_alt_name=dict(type='list', elements='str'),
+        eku=dict(type='str', choices=['SERVER_AUTH', 'CLIENT_AUTH', 'SERVER_AND_CLIENT_AUTH']),
+        ct_log=dict(type='bool'),
+        client_id=dict(type='int', default=1),
+        org=dict(type='str'),
+        ou=dict(type='list', elements='str'),
+        end_user_key_storage_agreement=dict(type='bool'),
+        tracking_info=dict(type='str'),
+        requester_name=dict(type='str'),
+        requester_email=dict(type='str'),
+        requester_phone=dict(type='str'),
+        additional_emails=dict(type='list', elements='str'),
+        custom_fields=dict(type='list', elements='dict', options=custom_fields_spec()),
+        cert_expiry=dict(type='str'),
+        cert_lifetime=dict(type='str', choices=['P1Y', 'P2Y', 'P3Y']),
+    )
+
+
+def main():
+    ecs_argument_spec = ecs_client_argument_spec()
+    ecs_argument_spec.update(ecs_certificate_argument_spec())
+    module = AnsibleModule(
+        argument_spec=ecs_argument_spec,
+        required_if=(
+            ['request_type', 'NEW', ['cert_type', 'requester_name', 'requester_email', 'requester_phone']],
+            ['cert_type', 'CODE_SIGNING', ['end_user_key_storage_agreement']],
+            ['cert_type', 'EV_CODE_SIGNING', ['end_user_key_storage_agreement']],
+        ),
+        mutually_exclusive=(
+            ['cert_expiry', 'cert_lifetime'],
+        ),
+        supports_check_mode=True,
+    )
+
+    # Does not yet support pyopenssl as a cryptography provider
```

Comment:
I wouldn't change that :)

(In particular with #59907 coming up ;) )

(And also, new pyOpenSSL versions have cryptography as a requirement.)

Reply:
> (And also, new pyOpenSSL versions have cryptography as a requirement.)

That's part of the reason I didn't bother, it was a pain setting up integration environments that had pyOpenSSL but not cryptography.

Reply:
We're still using it for the `openssl_pkcs12` module, since cryptography has no facilities for PKCS12 handling (even though the ones by pyOpenSSL are not very helpful as well...).

Also, the pyOpenSSL README has been saying for some time that one should not use it for anything other than making TLS connections: `Note: The Python Cryptographic Authority strongly suggests the use of pyca/cryptography where possible. If you are using pyOpenSSL for anything other than making a TLS connection you should move to cryptography and drop your pyOpenSSL dependency.` (https://github.com/pyca/pyopenssl/blob/master/README.rst)

Reply:
For 2 years now by the way: https://github.com/pyca/pyopenssl/issues/663